docs/concepts/workloads/containers/container-lifecycle-hooks.md

```diff
@@ -0,0 +1,163 @@
+
+* Information about the container itself.
+* Information about other objects in the system.
+
+### Container information
+
+The *hostname* of the container is the name of the pod in which the container is running. 
+It is available through the `hostname` command or the [`gethostname`](http://man7.org/linux/man-pages/man2/gethostname.2.html) function call in libc.
+
+The pod name and namespace are also available as environment variables through the [downward API](docs/tasks/configure-pod-container/downward-api-volume-expose-pod-information). 
+Additionally, user defined environment variables from the pod definition are also available to the container, 
+as are any environment variables specified statically in the Docker image.
+
+### Cluster information
+
+A list of all services that were running when a container was created via the Kubernetes Cluster API are available to that container as environment variables.
+The set of environment variables matches the syntax of Docker links.
+
+For a service named *foo* that maps to a container port named *bar*, 
+the following variables are defined:
+
+```shell
+FOO_SERVICE_HOST=<the host the service is running on>
+FOO_SERVICE_PORT=<the port the service is running on>
+```
+
+Services have dedicated IP addresses and are available to the container via DNS
+if [DNS addon](http://releases.k8s.io/{{page.githubbranch}}/cluster/addons/dns/) is enabled. 
+Because DNS is still not an enumerable protocol, 
+environment variables are provided so that Containers can do discovery.
+
+## Container hooks
+
+Container hooks provide information to the container about events in its management lifecycle.  
+For example, immediately after a container is started, it receives a `PostStart` hook.  
+Hooks are broadcast in the container with information about the lifecycle of the container.  
+They are different from the events provided by Docker or other systems which are output by the container.  
+Output events provide a log of what has already happened.  
+Input hooks, however, provide real-time notification about events that are happening, but without a historical log.
+
+### Hook details
+
+There are currently two Container hooks that are surfaced to containers:
+
+`PostStart`
+
+This hook is sent immediately after a container is created and notifies the container that it has been created.  
+No parameters are passed to the handler. 
+It is not guaranteed that the hook will execute before the container entrypoint.
+
+`PreStop`
+
+This hook is called immediately before a container is terminated. 
+No parameters are passed to the handler. 
+This event handler is blocking and must complete before the call to delete the container is sent to the Docker daemon. 
+The `SIGTERM` notification sent by Docker is also sent. 
+A more complete description of the termination behavior can be found in [Termination of Pods](/docs/user-guide/pods/#termination-of-pods).
+
+### Hook handler execution
+
+When a management hook occurs, 
```

Comment:
This could be simpler. "Immediately after a Container starts, Kubernetes calls the Container's PostStart handler."